example/z-docker-opam/Dockerfile

```diff
@@ -0,0 +1,32 @@
+FROM ocaml/opam:alpine as build
+
+# Install system dependencies
+# Some of these are only needed for database and JavaScript stuff, but this is 
+# our build context, it won't impact our runtime environment.
+RUN sudo apk add --update \
+    make libffi-dev linux-headers m4 ncurses-dev perl pkgconf postgresql-dev \
+    musl-dev git pkgconfig gmp-dev libc-dev openblas-dev zlib-dev \
+    npm nodejs libev-dev
+
+RUN eval $(opam env) && opam repository set-url default https://opam.ocaml.org && opam update
+
+WORKDIR /build
+
+# Install dependencies
+# We only copy the opam file to install dependencies only when they change, and 
+# not every time another file is modified.
+ADD hello.opam hello.opam
+RUN eval $(opam env) && opam install . --deps-only
+
+# Build project
+ADD . .
+
+RUN opam exec -- dune build --profile release
+
+FROM alpine:3.12 as run
```

Comment:
alpine quite often have some pretty bad CVE's that go un-patched. So a debian-slim variant is often a better pick.

Reply:
Oh! I wasn't aware of that, might try the `debian-slim` image, unfortunately there's no version of `ocaml/opam` with it, so we either have to install opam manually, or ask for a Debian image to be pushed to the `ocaml/opam` Docker repo

Reply:
@ulrikstrid Can you link to an article, summary, or example of or on this?

Reply:
http://crunchtools.com/comparison-linux-container-images/#Security

But mostly it's from what I've noticed when we used vurnability scanners.